### ISSUE TITLE
Type checking for TypeVar bound is not working when using type parameter syntax

### ISSUE THREAD

**Issue by Nugine** [microsoft/pyright#9074]
**Describe the bug**

Type checking for TypeVar bound is not working when using type parameter syntax ([PEP695](https://peps.python.org/pep-0695/)).

**Code or Screenshots**

```python
from typing import TypeVar, Generic


class Item:
    pass


T = TypeVar("T", bound=Item)


class Foo[T]:
    def get(self) -> T: ...


class Bar(Generic[T]):
    def get(self) -> T: ...


class SubFoo(Foo[int]):  # pass; should be error
    def get(self) -> int: ...


class SubBar(Bar[int]):  # error: Type "int" cannot be assigned to type variable "T@Bar"
    def get(self) -> int: ...

```

**VS Code extension or command-line**
Are you running pyright as a VS Code extension, a language server in another editor, integrated into Pylance, or the command-line tool? Which version?

+ command-line `pyright 1.1.381`
+ vscode extension `pylance v2024.9.2`

**Comment by erictraut** [comment]
You haven't provided an upper bound for `T` in the `Foo` class, so its upper bound defaults to `object`. To specify an upper bound, change your code to this:

```python
class Foo[T: Item]:
    def get(self) -> T: ...
```
